src/compiler/checker.ts

```diff
@@ -36090,7 +36094,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isGenericFunctionReturningFunction(signature: Signature) {
-        return !!(signature.typeParameters && isFunctionType(getReturnTypeOfSignature(signature)));
+        if (!signature.typeParameters) {
+            return false;
+        }
+        const returnType = getReturnTypeOfSignature(signature);
+        return returnType.flags & TypeFlags.Intersection ? someContainedType(returnType, isFunctionType) : isFunctionType(returnType);

```

Comment:
Isn't the point of the `someContainedType` helper that this can just be
```suggestion
        return someContainedType(returnType, isFunctionType);

```
?

Reply:
I intentionally dont want to run this for unions here. Not because I know we shouldnt but because I dont know if we should. Changing the behavior for unions would increase the likelihood of this breaking something for some project